src/lib/serviceRoutines/postQueryContext.cpp

```diff
@@ -86,6 +86,35 @@ static char* jsonPayloadClean(const char* payload)
 
 
 
+/* ***************************************************************************
+*
+* getProviderCount -
+*
+* getProviderCount is created for extrating the 'Fiware-Total-Count' header value so that we can add the same into total count,
+* And we can get the actualTotalCount value that includes total Entities of CP+CB.
+*
+*/
+void getProviderCount (std::string cpResponse, long long*  totalCount)
```

Comment:
Style

```suggestion
static void getProviderCount(std::string cpResponse, long long*  totalCount)
```

Reply:
Btw, this function is somehow a "hack" to get the value of a given header but maybe a more general solution would be to process all the headers in the response (in a `std:map`, for instance) so we can easily access to the `fiware-total-count` header just doing something like `reponseHeaders["fiware-total-count"]`.

In the same sense we have a function to get the response payload (I think is jsonPayloadClean()) we should have some responseHeaders() function for that. However, this can be deferred to a next PR, no need to include that in this (except if you think is easy and you want to do it :)

Reply:
We are investigating on this comment. As per your suggestion we will raise a separate PR for this comment.

Reply:
NTC (beyond the fix in https://github.com/telefonicaid/fiware-orion/commit/1e47b077ce86f15d09438331b764710751bf1910)